dynsettings/models.py

```diff
@@ -93,9 +126,10 @@ def get_value(cls, key, bucket=None):
 
         if not value:
             logger.info('Cache miss: {key}'.format(key=key))
-            # Dynamically add new value to db and reset cache
-            cls.add_key(key)
-            value = cls.load(key)
+            if key not in cls.valuedict:
+                logger.info('Dynsetting value missing from valuedict: {key}'.format(key=key))
+                cls.import_value_object(key)
+            value = cls._load(key)
```

Comment:
Debating replacing this with:

```suggestion
            value_object = cls.get_value_object(key)
            value = cls._load(key) or value_object.default_value
```